util/trxevents/trx_events.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trxevents
+
+import (
+	"unsafe"
+
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+)
+
+// EventType represents the type of a transaction event.
+type EventType = int
+
+const (
+	// EventTypeCopMeetLock stands for the CopMeetLock event type.
+	EventTypeCopMeetLock = iota
+)
+
+// CopMeetLock represents an event that coprocessor reading encounters lock.
+type CopMeetLock struct {
+	LockInfo *kvrpcpb.LockInfo
+}
+
+// TransactionEvent represents a transaction event that may belong to any of the possible types.
+type TransactionEvent struct {
+	eventType EventType
+	ptr       unsafe.Pointer
```

Comment:
Why we have to use `unsafe.Pointer` here?

Reply:
It seems either can be used, but interface{} looks better.  I thought about avoiding the RTTI (though I'm not sure if there is) but I think perhaps that's a bad idea. I'll change it to interface{} then, please wait a minute